### PR TITLE
feat(node): support legacy transactions

### PR DESCRIPTION
**Pull request by cryptoAtwill**
Closes https://github.com/consensus-shipyard/ipc/issues/927.

The current main does not allow `send_raw_transaction` for legacy transactions. This PR removes this assumption. To achieve it, the following changes are made:
- Adding `OriginKind` enum to explicitly specify the original transaction type. The type information is lost when converting to Fvm messages.
- Support gas params from legacy transactions as specified in https://github.com/filecoin-project/FIPs/blob/master/FIPS/fip-0091.md
- Any ethereum transaction sent from fendermint client is enforced to be eip1559.
- `TransactionCache` is updated to use `(TypeTransaction, Signature)` instead of the response dto `Transaction`.

### CODE REVIEW

File: fendermint/eth/api/src/conv/from_eth.rs
Comment (raulk) [inline fendermint/eth/api/src/conv/from_eth.rs:92]: Why are we duplicating this logic here? `send_raw_transaction` is already producing this error?

File: fendermint/eth/api/src/apis/eth.rs
Comment (raulk) [inline fendermint/eth/api/src/apis/eth.rs:645]: What do you mean with transaction _response_ here? This module is an API, so I interpret as if this is returning the response the API sends to the caller, which isn't the case? (unless there's an error)

We are duplicating the transaction type validation in this method and when we calculate the origin kind above. Can we refactor this to make it DRY?

BTW -- how can we have a module called `from_eth` that then has `to_eth` methods? Wouldn't they cancel each other out? 😆

File: fendermint/eth/api/src/conv/from_eth.rs
Comment (raulk) [inline fendermint/eth/api/src/conv/from_eth.rs:31]: I really dislike that the cache uses `et::Transaction` -- this type is intended as a return type from the JSON-RPC API, not to store canonical signed transactions.

File: fendermint/eth/api/src/conv/from_eth.rs
Comment (raulk) [inline fendermint/eth/api/src/conv/from_eth.rs:32]: This all looks very repetitive. Can't you populate the common values from `TransactionEssentials` via [`essentials()`](https://github.com/foundry-rs/foundry/blob/eef87de35375311f134fb4261c18455ced8022a5/crates/anvil/core/src/eth/transaction/mod.rs#L690)? And then add the specifics?

Isn't there a method to normalize typed transactions into the Transaction response type in ethers?

**Review by raulk — CHANGES_REQUESTED** [review]

File: fendermint/eth/api/src/conv/from_eth.rs
Comment (cryptoAtwill) [reply fendermint/eth/api/src/conv/from_eth.rs:32]: > Isn't there a method to normalize typed transactions into the Transaction response type in ethers?
From what I found, not really. My guess is the Transaction response contains transaction index, which does not make sense to do the conversion using `From`.

**Comment by cryptoAtwill** [comment]
@raulk Because the types and `from_eth::*` and `to_eth::*` are used in multiple places, I just did a extension to make it work. I think it's a good chance to do a more in depth refactoring. Will update accordingly.

File: fendermint/eth/api/src/mpool.rs
Comment (raulk) [inline fendermint/eth/api/src/mpool.rs:28]: Honestly, now that I look at the final solution, I think this was better the way it was before. We ended up leaking all of ethers' type complexity into Tendermint. In fact, I don't think TypedTransaction should've leaked anywhere at all. I'll work on rolling this back, removing `to_eth_typed_transaction`, `to_eth_legacy_request`, `to_eth_eip1559_request`, etc.

File: fendermint/eth/api/src/apis/eth.rs
Comment (raulk) [reply fendermint/eth/api/src/apis/eth.rs:618]: Looks like it's not exposed...

File: fendermint/eth/api/src/apis/eth.rs
Comment (cryptoAtwill) [reply fendermint/eth/api/src/apis/eth.rs:445]: they should be the same, `debug_assert_eq` should be fine for release build though, more like a sanity check for tests.

File: fendermint/eth/api/src/apis/eth.rs
Comment (cryptoAtwill) [reply fendermint/eth/api/src/apis/eth.rs:618]: Yeah, a lot of methods are not exposed. Could be helpful util tools.

File: fendermint/eth/api/src/apis/eth.rs
Comment (cryptoAtwill) [reply fendermint/eth/api/src/apis/eth.rs:634]: It won't remain forever as it's using LRU cache. It will just stay for some time, depending on the configuration.

File: fendermint/eth/api/src/mpool.rs
Comment (cryptoAtwill) [reply fendermint/eth/api/src/mpool.rs:28]: I feel the `conv::*` should not be in `message` crate. It should be shifted to `eth` crate. That might be better.

**Review by raulk — APPROVED** [review]
Let's go ahead and merge this to unblock users who are waiting for this feature to land. We can clean this up in main later.

**Pull request merged**
